src/session.rs

```diff
@@ -2801,6 +2801,9 @@ impl Session {
             Command::PaletteClear => {
                 self.palette.clear();
             }
+            Command::PaletteGradient(colorstart, colorend, steps) => {
+                self.palette.gradient(colorstart, colorend, steps)
```

Comment:
You're just missing a `self.center_palette()` here to re-center the palette after the colors are added.

Reply:
Thanks. It is done.